src/objects/zif_abapgit_object.intf.abap

```diff
@@ -25,46 +26,80 @@ INTERFACE zif_abapgit_object
       !iv_transport TYPE trkorr
     RAISING
       zcx_abapgit_exception .
+
   METHODS delete
     IMPORTING
       !iv_package   TYPE devclass
       !iv_transport TYPE trkorr
     RAISING
       zcx_abapgit_exception .
+
   METHODS exists
     RETURNING
       VALUE(rv_bool) TYPE abap_bool
     RAISING
       zcx_abapgit_exception .
+
   METHODS is_locked
     RETURNING
       VALUE(rv_is_locked) TYPE abap_bool
     RAISING
       zcx_abapgit_exception .
+
   METHODS is_active
     RETURNING
       VALUE(rv_active) TYPE abap_bool
     RAISING
       zcx_abapgit_exception .
+
   METHODS changed_by
+    IMPORTING
+      !is_sub_item   TYPE zif_abapgit_definitions=>ty_item OPTIONAL
     RETURNING
       VALUE(rv_user) TYPE syuname
     RAISING
       zcx_abapgit_exception .
+
   METHODS jump
+    IMPORTING
+      !is_sub_item   TYPE zif_abapgit_definitions=>ty_item OPTIONAL
     RETURNING
       VALUE(rv_exit) TYPE abap_bool
     RAISING
       zcx_abapgit_exception .
+
   METHODS get_metadata
     RETURNING
       VALUE(rs_metadata) TYPE zif_abapgit_definitions=>ty_metadata .
+
   METHODS get_comparator
     RETURNING
       VALUE(ri_comparator) TYPE REF TO zif_abapgit_comparator
     RAISING
       zcx_abapgit_exception .
+
   METHODS get_deserialize_steps
     RETURNING
       VALUE(rt_steps) TYPE zif_abapgit_definitions=>ty_deserialization_step_tt .
+
+  METHODS get_deserialize_order
+    IMPORTING
+      !it_all_objects          TYPE zif_abapgit_definitions=>ty_items_tt
+    RETURNING
+      VALUE(rt_objects_before) TYPE zif_abapgit_definitions=>ty_items_tt.
+
+  METHODS map_filename_to_object
```

Comment:
wondering, should this method be static? it takes a filename to get the object, but it requires the object to call the constructor?

Reply:
Technically it should be but then the method can't be part of the interface.

Reply:
below works, its strange to have static methods part of an interface, but I think its possible in ABAP

and either way, how to call the method? supply dummy values to the instantiation/constructor?

```abap
INTERFACE lif.
  CLASS-METHODS foo.
ENDINTERFACE.
```

Reply:
Oh yes. Forgot about that. I will change it